src/main/java/org/sagebionetworks/bridge/spring/controllers/AdherenceController.java

```diff
@@ -244,4 +245,19 @@ public StatusMessage deleteAdherenceRecord(
         service.deleteAdherenceRecord(record);
         return DELETED_MSG;
     }
+    
+    @GetMapping("/v5/studies/{studyId}/participants/{userId}/adherence/detail")
+    public DetailedAdherenceReport getDetailedParticipantAdherenceReport(@PathVariable String studyId,
+                                                                         @PathVariable String userId) {
+        UserSession session = getAuthenticatedSession(DEVELOPER, RESEARCHER, STUDY_DESIGNER, STUDY_COORDINATOR);
+        
+        CAN_READ_STUDIES.checkAndThrow(STUDY_ID, studyId);
```

Comment:
I think for Adherence, we usually use CAN_ACCESS_ADHERENCE_DATA. This takes in both studyId and userId (although since we never get adherence for self in this API, the userId won't actually matter).

Unrelated, it appears that the other adherence report APIs in this Controller don't check that the caller has access to the study.

Reply:
Updated. It looks like the other reports are doing their auth checks in the service instead of the controller.